highlight/highlight
# Waterfall view for a trace

## Summary

Adds a waterfall view of a trace alongside the flame graph. The waterfall view allows you to see all spans in a list and provides a visual of the duration a span takes on the right. It also allows you to filter the spans using the input at the top to quickly find a span of interest.

<img width="950" alt="Screenshot 2024-04-30 at 4 10 31 PM" src="https://github.com/highlight/highlight/assets/308182/a972106f-63bc-4dfd-aec2-3530614ed474">

This PR also fixes an issue with orphan spans not being rendered on certain traces ([HIG-4569](https://linear.app/highlight/issue/HIG-4569/trace-flame-graph-not-showing-all-spans)).

## How did you test this change?

* Go to a span anywhere in the product
* Click the "Waterfall" tab to switch from the flame graph to the waterfall view
* Ensure the functionality of the visualization behaves as expected

Please note that lots of the noise in the diff is from a change to our sprinkles and the generated files.

## Are there any deployment considerations?

N/A - all client-side changes.

## Does this work require review from our design team?

Julian has already provided a round of feedback and provided mocks, but will have him click test as well!
